netket/utils/mpi/util.py

```diff
@@ -0,0 +1,26 @@
+import jax
+
+
+def mpi_tree_map(mpi_fun, x, *, token=None):
```

Comment:
Maybe this should just go into `primitives.py`?

Reply:
in my wicked mind primitives.py is for mpi primitives. This is really an util to enforce an order when tree_mapping... I'm not even sure we'll ever do something complex enough where it's needed, but I wrote it when tracking down some bugs so I thought it might be useful to leave it here.

Reply:
but maybe I'll delete it

Reply:
Fair enough, it's not a primitive indeed. I just felt that `netket.utils.mpi.util` is a bit too much `util`ity for one module path.

Reply:
I'll remove it. No point in having dead code unused everywhere

Reply:
done. this should be ok now.